docs/design/interpreter/compiled-code-interop.md

```diff
@@ -0,0 +1,69 @@
+### Compiled code interop
+
+The interpreter is expected to run in combination with aot compiled code so we will need an efficient mechanism for entering/exiting interpreter. Even in an interp only mode, we would still need to do these transitions for pinvokes and reverse pinvokes. These transitions also pose a few challenges since on ios we won't be able to dynamically generate thunks, while on WASM the call signature must be embedded into the emitted wasm code, meaning we can't reuse a generic thunk for calling methods with different signatures.
```

Comment:
```suggestion
The interpreter is expected to run in combination with aot compiled code so we will need an efficient mechanism for entering/exiting interpreter. Even in an interp only mode, we would still need to do these transitions for pinvokes and reverse pinvokes. These transitions also pose a few challenges since on iOS we won't be able to dynamically generate thunks, while on Wasm the call signature must be embedded into the emitted Wasm code, meaning we can't reuse a generic thunk for calling methods with different signatures.
```